pkg/lib/bundle/generate.go

```diff
@@ -93,19 +98,29 @@ func GenerateFunc(directory, packageName, channels, channelDefault string, overw
 	return nil
 }
 
-// GenerateFunc determines mediatype from files (yaml) in given directory
+// GetMediaType determines mediatype from files (yaml) in given directory
 // Currently able to detect helm chart, registry+v1 (CSV) and plain k8s resources
 // such as CRD.
 func GetMediaType(directory string) (string, error) {
-	var files []string
+	files := make(map[string][]byte)
 
 	// Read all file names in directory
 	items, _ := ioutil.ReadDir(directory)
 	for _, item := range items {
 		if item.IsDir() {
 			continue
-		} else if filepath.Ext(item.Name()) == ".yaml" {
-			files = append(files, item.Name())
+		}
+
+		filewithpath := filepath.Join(directory, item.Name())
+		fileblob, err := ioutil.ReadFile(filewithpath)
+		if err != nil {
+			return "", fmt.Errorf("Unable to read file %s in bundle", filewithpath)
+		}
+
+		var yamlfile map[string]interface{}
+		err = yaml.Unmarshal(fileblob, &yamlfile)
```

Comment:
👍

Reply:
There's also a "yamlOrJson" decoder somewhere that we can use - don't see any reason to exclude json